### PR TITLE
Materialize Flutter module, Android

### PR DESCRIPTION
**Pull request by mravn-google**

### CODE REVIEW

**Review by sigurdm — COMMENTED** [review]
I'm a bit confused about the need for differences between the ephemeral and the materialized template - I was hoping the two apps could be the same, just at different locations.

**Review by mravn-google — COMMENTED** [review]
Thanks for the review

File: packages/flutter_tools/templates/module/android_host_materialized/gradle.properties.copy.tmpl [hunk 1]
```diff
@@ -0,0 +1 @@
+org.gradle.jvmargs=-Xmx1536M
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply packages/flutter_tools/templates/module/android_host_materialized/gradle.properties.copy.tmpl]: Sorry - I read the change wrong.

File: packages/flutter_tools/test/project_test.dart [hunk 2]
```diff
@@ -325,6 +352,24 @@ void transfer(FileSystemEntity entity, FileSystem target) {
...
+  } on ToolExit catch(e) {
+    expect(e.message, messageMatcher);
+  } catch(e, trace) {
+    fail('ToolExit expected, got $e\n$trace');
+  }
+}
+
+void expectExists(FileSystemEntity entity) {
+  expect(entity.existsSync(), isTrue);
+}
```
Comment (sigurdm) [inline packages/flutter_tools/test/project_test.dart]: Newline between functions

**Review by sigurdm — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/project_test.dart (hunk 2 above)
Comment (mravn-google) [reply packages/flutter_tools/test/project_test.dart]: Done.

**Review by mravn-google — COMMENTED** [review]
Refactored the android templates to improve structure and have less duplication. Fixed error in materialize (no local.properties generated for host app) and its device lab test (flutter packages get not invoked after adding plugins).

PTAL

**Review by sigurdm — APPROVED** [review]
LGTM

**Pull request merged**